### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by b97pla**
- Some documentation updates
- Changed inputs to export and extract subcommands
- Added some tests (still a lot of broken tests after refactoring)
-

### CODE REVIEW

File: snpseq_metadata/models/converter.py [hunk 1]
```diff
@@ -50,7 +59,17 @@ class Converter:
     def ngi_to_sra(
         cls: Type[T], ngi_model: ngi_model_class
     ) -> Optional[sra_model_class]:
-        # only recurse if this is called in the base class
+        """
+        Entry point to convert a NGI model class to a corresponding SRA model class. The method will
+        iterate over subclasses to locate a subclass whose ngi_model_class matches the supplied
+        ngi_model.
```
Comment (GitEdvard) [inline snpseq_metadata/models/converter.py:65]: What if there is no one-to-one match between ngi-,sra-, or lims-models?

File: snpseq_metadata/models/converter.py [hunk 2]
```diff
@@ -62,7 +81,17 @@ def ngi_to_sra(
     def lims_to_ngi(
         cls: Type[T], lims_model: lims_model_class
     ) -> Optional[ngi_model_class]:
-        # only recurse if this is called in the base class
+        """
+        Entry point to convert a LIMS model class to a corresponding NGI model class. The method
+        will iterate over subclasses to locate a subclass whose lims_model_class matches the
+        supplied lims_model.
```
Comment (GitEdvard) [inline snpseq_metadata/models/converter.py:87]: I wonder about Lims models contains the class sequencing_container, which have no counterpart in ngi models.

File: snpseq_metadata/models/converter.py [hunk 3]
```diff
@@ -41,6 +41,15 @@
...
 
 class Converter:
+    """
+    The main class for doing conversions between models. The idea is that the conversion is made by
+    calling the appropriate class method and supplying the model instance to convert from. This
+    pattern allows the models to be implemented independently of each other, i.e. the model modules
+    themselves need not know anything about any other model.
+
+    Example:
+        sra_library_instance = Converter.ngi_to_sra(ngi_library_instance)
```
Comment (GitEdvard) [inline snpseq_metadata/models/converter.py:51]: I'm not sure what's the meaning of ngi-models. Is it the models used in the genologics repository?

**Review by GitEdvard — COMMENTED** [review]
It is always nice to see code from a new person, with new coding styles. Great work. I have some questions though. 1) I think that the purpose of the xs data classes is to create outputs in xml format. Is it so? If so, I think that it should be written in the docs. 2) I'm a bit unsure about the meaning of "NGI models". 3) I can't get the tests to work (~50% of them passes).

**Comment by GitEdvard** [comment]
Now, I red the doc another time, and I see that the xsData is included here. As I understand it, some of the code is auto-generated. Is it all thing under xsData? Perhaps, there should be some indication of which of the classes are generated by xsData?

File: snpseq_metadata/models/converter.py (hunk 1 above)
Comment (b97pla) [reply snpseq_metadata/models/converter.py:65]: hmm.. well the idea is that the converter class should handle the logic of converting between the model systems. If there is no corresponding conversion that could be made, it is expected to return None

File: snpseq_metadata/models/converter.py (hunk 3 above)
Comment (b97pla) [reply snpseq_metadata/models/converter.py:51]: well, the naming of things may not be the best.. I started out with the NGI models meant to represent the metadata as we, NGI, see it. The sources for this data are, as you know, a mix between clarity and Illumina run folder output. On top of that came the LIMSmodel which is more directly tied to clarity and what your service exports.

**Comment by b97pla** [comment]
Thanks for your valuable comments Edvard! Having worked some more on this and the tests, I think more work is needed before it is ready to merge. So let's leave this as it is for now!

**Comment by b97pla** [comment]
@GitEdvard I've been doing some work on this PR and there is now an extensive unit test suite as well as some integration tests with example data (these are run by travis as well).
I realize that it's probably not realistic that you review this extensively and I'm ok with merging myself as is. So feel free to browse around and comment to the extent that you feel is manageable and let me know if you want me to merge or discuss any particulars.

**Pull request merged**